src/consumption.cpp

```diff
@@ -1612,19 +1610,31 @@ bool Character::can_consume( const item &it ) const
     if( can_consume_as_is( it ) ) {
         return true;
     }
-    // Checking NO_RELOAD to prevent consumption of `battery` when contained in `battery_car` (#20012)
-    return !it.is_container_empty() && !it.has_flag( flag_NO_RELOAD ) &&
-           can_consume_as_is( it.contents.front() );
+    return has_item_with( [&]( const item & consumable ) {
+        // Checking NO_RELOAD to prevent consumption of `battery` when contained in `battery_car` (#20012)
+        return !consumable.has_flag( flag_NO_RELOAD ) && can_consume_as_is( consumable );
+    } );
 }
 
 item &Character::get_consumable_from( item &it ) const
 {
-    if( !it.is_container_empty() && can_consume_as_is( it.contents.front() ) ) {
-        return it.contents.front();
-    } else if( can_consume_as_is( it ) ) {
+    if( can_consume_as_is( it ) ) {
         return it;
     }
 
+    item *ret = nullptr;
+    it.visit_items( [&]( item * it ) {
```

Comment:
small note: since visit_items() visits itself first, the if statemnt above can be removed.